apps/studio/components/to-be-cleaned/Storage/StoragePolicies/StoragePolicies.tsx

```diff
@@ -1,37 +1,64 @@
 import { filter, find, get, isEmpty } from 'lodash'
 import { observer } from 'mobx-react-lite'
-import { useEffect, useState } from 'react'
+import { useState } from 'react'
 import { IconLoader } from 'ui'
 
 import ConfirmModal from 'components/ui/Dialogs/ConfirmDialog'
 import { useStore } from 'hooks'
-import { useStorageStore } from 'localStores/storageExplorer/StorageExplorerStore'
 import { formatPoliciesForStorage } from '../Storage.utils'
 import StoragePoliciesBucketRow from './StoragePoliciesBucketRow'
 import StoragePoliciesEditPolicyModal from './StoragePoliciesEditPolicyModal'
 import StoragePoliciesPlaceholder from './StoragePoliciesPlaceholder'
 
 import { useParams } from 'common'
 import { PolicyEditorModal } from 'components/interfaces/Auth/Policies'
+import { useProjectContext } from 'components/layouts/ProjectLayout/ProjectContext'
+import { useDatabasePoliciesQuery } from 'data/database-policies/database-policies-query'
 import { useBucketsQuery } from 'data/storage/buckets-query'
+import { useDatabasePolicyCreateMutation } from 'data/database-policies/database-policy-create-mutation'
+import { useDatabasePolicyUpdateMutation } from 'data/database-policies/database-policy-update-mutation'
+import { useDatabasePolicyDeleteMutation } from 'data/database-policies/database-policy-delete-mutation'
+import toast from 'react-hot-toast'
 
 const StoragePolicies = () => {
   const { ui, meta } = useStore()
+  const { project } = useProjectContext()
   const { ref: projectRef } = useParams()
 
-  const storageStore = useStorageStore()
-  const { loaded } = storageStore
-
-  const { data } = useBucketsQuery({ projectRef })
+  const { data, isLoading: isLoadingBuckets } = useBucketsQuery({ projectRef })
   const buckets = data ?? []
 
   const roles = meta.roles.list((role: any) => !meta.roles.systemRoles.includes(role.name))
 
-  const [policies, setPolicies] = useState<any[]>([])
   const [selectedPolicyToEdit, setSelectedPolicyToEdit] = useState({})
   const [selectedPolicyToDelete, setSelectedPolicyToDelete] = useState<any>({})
   const [isEditingPolicyForBucket, setIsEditingPolicyForBucket] = useState<any>({})
 
+  const {
+    data: policiesData,
+    refetch,
+    isLoading: isLoadingPolicies,
+  } = useDatabasePoliciesQuery({
+    projectRef: project?.ref,
+    connectionString: project?.connectionString,
+    schema: 'storage',
+  })
+  const policies = policiesData ?? []
+
+  const isLoading = isLoadingBuckets || isLoadingPolicies
+
+  const { mutateAsync: createDatabasePolicy } = useDatabasePolicyCreateMutation({
+    onError: () => {},
```

Comment:
yup this is to suppress the default error handling since we're handling it below in line 125
we're handling there as well cause we need it to be async so that the policy modal only closes after _all_ policies are created